cli/azd/cmd/show.go

```diff
@@ -0,0 +1,188 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/internal"
+	"github.com/azure/azure-dev/cli/azd/pkg/commands"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/infra"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/spf13/cobra"
+)
+
+func showCmd(rootOptions *internal.GlobalCommandOptions) *cobra.Command {
+	action := func(ctx context.Context, cmd *cobra.Command, args []string, azdCtx *azdcontext.AzdContext) error {
+		console := input.GetConsole(ctx)
+
+		formatter := output.GetFormatter(ctx)
+		writer := output.GetWriter(ctx)
+
+		// Right now this command is hidden and we only expect it to be called by tooling,
+		// which passes `--output json`. If for some reason someone ran it directly, just
+		// don't do anything.
+		if formatter.Kind() != output.JsonFormat {
+			return nil
+		}
+
+		if err := ensureProject(azdCtx.ProjectPath()); err != nil {
+			return err
+		}
+
+		env, ctx, err := loadOrInitEnvironment(ctx, &rootOptions.EnvironmentName, azdCtx, console)
+		if err != nil {
+			return fmt.Errorf("loading environment: %w", err)
+		}
+
+		prj, err := project.LoadProjectConfig(azdCtx.ProjectPath(), &env)
+		if err != nil {
+			return fmt.Errorf("loading project: %w", err)
+		}
+
+		res := showResult{
+			Services: make(map[string]showService, len(prj.Services)),
+		}
+
+		for name, svc := range prj.Services {
+			path, err := getFullPathToProjectForService(svc)
+			if err != nil {
+				return err
+			}
+
+			showSvc := showService{
+				Project: showServiceProject{
+					Path: path,
+					Type: showTypeFromLanguage(svc.Language),
+				},
+			}
+
+			res.Services[name] = showSvc
+		}
+
+		// Add information about the target of each service, if we can determine it (if the infrastructure has
+		// not been deployed, for example, we'll just not include target information)
+		resourceManager := infra.NewAzureResourceManager(ctx)
+		if resourceGroupName, err := resourceManager.FindResourceGroupForEnvironment(ctx, &env); err == nil {
+			for name := range prj.Services {
+				if resources, err := project.GetServiceResources(ctx, resourceGroupName, name, &env); err == nil {
+					if len(resources) == 1 {
+						resSvc := res.Services[name]
+						resSvc.Target = &showTargetArm{
+							ResourceId: resources[0].Id,
+						}
+						res.Services[name] = resSvc
+					}
+				} else {
+					log.Printf("ignoring error determining resource id for service %s: %v", name, err)
```

Comment:
Trying to determine the severity of this. What happens if we don't populate `target` here due to a failure in resolving the target resource? Note that this does not mean the service isn't deployed -- it means connection to Azure is broken. Does this lead to unexpected behavior?